plugins/com.google.cloud.tools.eclipse.usagetracker.test/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManagerTest.java

```diff
@@ -178,10 +153,62 @@ public void testSendPingScheduled_optInRegisteredAndAlreadyOptedIn() {
 
   private void verifyPingQueued(VerificationMode verificationMode) {
     when(pingEventQueue.isEmpty()).thenReturn(true);
-    AnalyticsPingManager pingManager =
-        new AnalyticsPingManager(preferences, display, pingEventQueue, true);
-    pingManager.unitTestMode = true;
     pingManager.sendPing("eventName", "metadataKey", "metadataValue");
     verify(pingEventQueue, verificationMode).add(any(PingEvent.class));
   }
+
+  @Test
+  public void testGetAnonymizedClientId_generateNewId() {
+    when(preferences.get(eq(AnalyticsPreferences.ANALYTICS_CLIENT_ID), anyString()))
+        .thenReturn(null);  // Simulate that client ID has never been generated.
+    String clientId = pingManager.getAnonymizedClientId();
+    assertTrue(!clientId.isEmpty());
+    verify(preferences, times(1)).put(eq(AnalyticsPreferences.ANALYTICS_CLIENT_ID), eq(clientId));
```

Comment:
using `times(1)` is optional

also you may be able to remove the `eq()` calls when using actual values (instead of mixing values and matchers like `any()`)